docs/tutorials/mastering.md

```diff
@@ -143,68 +133,257 @@ scp /tmp/control-plane-cluster.yaml juju-controller.mgmt:
+sudo apt update
+sudo apt install -y driverctl
+```
+
+Bind `access` and `core` interfaces to `vfio-pci` driver persistently:
+
+```console
+cat << EOF | sudo tee -a /etc/rc.local
+#!/bin/bash
+sudo driverctl set-override 0000:07:00.0 vfio-pci
+sudo driverctl set-override 0000:06:00.0 vfio-pci
 EOF
 sudo chmod +x /etc/rc.local
 sudo /etc/rc.local
 ```
 
+##### Checkpoint 3: Validate that the VFIO-PCI driver is loaded
+
+Check the current driver of interfaces by running the following command:
+
+```console
+sudo driverctl -v list-devices | grep -i net
+```
+
+You should see the following output:
+
+```
+0000:05:00.0 virtio-pci (Virtio network device)
+0000:06:00.0 vfio-pci [*] (Virtio network device)
+0000:07:00.0 vfio-pci [*] (Virtio network device)
+```
+
+Verify that two VFIO devices are created with a form of `noiommu-{a number}` by running the following command:
+
+```console
+ls -l /dev/vfio/
+```
+
+You should see a similar output:
+
+```
+crw------- 1 root root 242,   0 Aug 17 22:15 noiommu-0
+crw------- 1 root root 242,   1 Aug 17 22:16 noiommu-1
+crw-rw-rw- 1 root root  10, 196 Aug 17 21:51 vfio
+```
+
+#### Install Kubernetes Cluster
+
+Install MicroK8s with the necessary plugins:
+
+```console
+sudo snap install microk8s --channel=1.31/stable --classic
+sudo microk8s enable hostpath-storage
+sudo microk8s addons repo add community https://github.com/canonical/microk8s-community-addons --reference feat/strict-fix-multus
+sudo microk8s enable multus
+sudo usermod -a -G microk8s $(whoami)
+sudo snap alias microk8s.kubectl kubectl
+sudo microk8s enable metallb:10.201.0.200/32
+sudo microk8s disable dns
+sudo microk8s enable dns:10.201.0.1
+newgrp microk8s
+```
+
+#### Configure Kubernetes for DPDK
+
+Create [SR-IOV Network Device Plugin] ConfigMap by replacing the `pciAddresses` with the PCI addresses of `access` and `core` interfaces:
+
+```console
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: sriovdp-config
+  namespace: kube-system
+data:
+  config.json: |
+    {
+      "resourceList": [
+        {
+          "resourceName": "intel_sriov_vfio_access",
+          "selectors": {
+            "pciAddresses": ["0000:07:00.0"]
+          }
+        },
+        {
+          "resourceName": "intel_sriov_vfio_core",
+          "selectors": {
+            "pciAddresses": ["0000:06:00.0"]
+          }
+        }
+      ]
+    }
+
+EOF
+```
+
+Deploy [SR-IOV Network Device Plugin]:
+
+```console
+kubectl apply -f https://raw.githubusercontent.com/k8snetworkplumbingwg/sriov-network-device-plugin/v3.6.2/deployments/sriovdp-daemonset.yaml
+```
+
+##### Checkpoint 4: Check the allocatable resources in the Kubernetes node
+
+Make sure that there are 2 `1Gi HugePages`, 1 `intel_sriov_vfio_access` and 1 `intel_sriov_vfio_core` are available by running the following command:
+
+```console
+sudo snap install jq
+kubectl get node -o json | jq '.items[].status.allocatable'
+```
+
+You should see the following output:
```

Comment:
Maybe we can include a note that this might not happen immediately to avoid confusing readers if it takes a bit of time?

Reply:
Done